addons/account/models/account_move.py

```diff
@@ -1674,34 +1674,32 @@ def _check_balanced(self, container):
         # /!\ As this method is called in create / write, we can't make the assumption the computed stored fields
         # are already done. Then, this query MUST NOT depend of computed stored fields (e.g. balance).
         # It happens as the ORM makes the create with the 'no_recompute' statement.
-        self.env['account.move.line'].flush_model(['balance', 'currency_id', 'move_id'])
+        self.env['account.move.line'].flush_model(['balance', 'credit', 'debit', 'currency_id', 'move_id'])
         self.env['account.move'].flush_model(['journal_id'])
         self._cr.execute('''
-            WITH error_moves AS (
-                SELECT line.move_id,
-                       ROUND(SUM(line.balance), currency.decimal_places) balance
-                  FROM account_move_line line
-                  JOIN account_move move ON move.id = line.move_id
-                  JOIN account_journal journal ON journal.id = move.journal_id
-                  JOIN res_company company ON company.id = journal.company_id
-                  JOIN res_currency currency ON currency.id = company.currency_id
-                 WHERE line.move_id IN %s
-              GROUP BY line.move_id, currency.decimal_places
-            )
-            SELECT *
-              FROM error_moves
-             WHERE balance !=0
+            SELECT line.move_id,
+                   ROUND(SUM(line.debit), currency.decimal_places) debit,
+                   ROUND(SUM(line.credit), currency.decimal_places) credit
+              FROM account_move_line line
+              JOIN account_move move ON move.id = line.move_id
+              JOIN account_journal journal ON journal.id = move.journal_id
+              JOIN res_company company ON company.id = journal.company_id
+              JOIN res_currency currency ON currency.id = company.currency_id
+             WHERE line.move_id IN %s
+          GROUP BY line.move_id, currency.decimal_places
+            HAVING ROUND(SUM(line.balance), currency.decimal_places) != 0   
```

Comment:
@vin-odoo Why not use debit-credit then here? (and it is contrary to the comment above as well)

Reply:
@epictete Should not we change the comment then?

Reply:
Indeed, the comment is still valid, but the example is not anymore :wink:

Reply:
I've removed the example of the comment :+1: